enhancements/network/user-defined-network-segmentation.md

```diff
@@ -379,11 +382,283 @@ any user-defined primary network.
+| Topology       | The topological configuration for the network. Must be one of `Layer2`, `layer3`, `Localnet`.                                                                                                                                                                                                                                                                                                          | No       |
+| Role           | Select the network role in the pod, either `Primary` or `Secondary`. Primary network support topologies `Layer2` and `Layer3` only.                                                                                                                                                                                                                                                                    | No       |
+| MTU            | The maximum transmission unit (MTU).<br/>The default is 1400.                                                                                                                                                                                                                                                                                                                                          | Yes      |
+| Subnets        | The subnet to use for the network across the cluster.<br/>E.g. `10.100.200.0/24`.<br/>IPv6 `2001:DBB::/64` and dual-stack `192.168.100.0/24`,`2001:DBB::/64` subnets are supported.<br/>When omitted, the logical switch implementing the network only provides layer 2 communication, and users must configure IP addresses.<br/>Port security only prevents MAC spoofing if the subnets are omitted. | Yes      |
+| ExcludeSubnets | List of CIDRs.<br/>IP addresses are removed from the assignable IP address pool and are never passed to the pods.                                                                                                                                                                                                                                                                                      | Yes      |
+| JoinSubnets    | Subnet used inside the OVN network topology.  When omitted, this means no opinion and the platform is left to choose a reasonable default which is subject to change over time.                                                                                                                                                                                                                        | Yes      |
+| IPAMLifecycle  | Control IP addresses management lifecycle. When `Persistent` is specified it enable workloads have persistent IP addresses. For example: Virtual Machines will have the same IP addresses along their lifecycle (stop, start migration, reboots). Supported by Topology `Layer2` & `Localnet`.                                                                                                         | Yes      |
```

Comment:
I think you want a discriminated union as part of this API, given you have multiple options that only support some of the topologies

Reply:
Since the `IPAMLifecycle`  behavior is the same for both `layer2` and `localnet` typologies, I didnt add additional union that specify the topology, make sense?